src/Diagnostics.ModelsAndUtils/Models/GeoMaster/AzureSupportCenterInsight.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Diagnostics.ModelsAndUtils.Attributes;
+using Diagnostics.ModelsAndUtils.Models.ResponseExtensions;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Models.GeoMaster
+{
+    public class AzureSupportCenterInsight
+    {
+        public Guid Id { get; set; }
+
+        public string Title { get; set; }
+
+        public ImportanceLevel ImportanceLevel { get; set; }
+
+        public string InsightFriendlyName { get; set; }
+
+        public string IssueCategory { get; set; }
+
+        public string IssueSubCategory { get; set; }
+
+        public Text Description { get; set; }
+
+        public RecommendedAction RecommendedAction { get; set; }
+
+        public CustomerReadyContent CustomerReadyContent { get; set; }
+
+        public IEnumerable<Link> Links { get; set; }
+
+        public string ConfidenceLevel { get; set; }
+
+        public string Scope { get; set; }
+    }
+
+    public enum ImportanceLevel
+    {
+        SomeLevel
+    }
+
+    public class CustomerReadyContent
+    {
+        public string ArticleId { get; set; }
+
+        public Text Text { get; set; }
+    }
+
+    public class Text
+    {
+        public bool IsMarkdown { get; set; }
+
+        public string Value { get; set; }
+    }
+
+    public class Link
+    {
+        public int Type { get; set; }
+
+        public string Text { get; set; }
+
+        public string Uri { get; set; }
+    }
+
+    public class RecommendedAction
+    {
+        public Guid Id { get; set; }
+
+        public Text Text { get; set; }
+    }
+
+    public static class AzureSupportCenterInsightUtilites
+    {
+        public static AzureSupportCenterInsight CreateInsight<TResource>(Insight insight, OperationContext<TResource> context, Definition detector)
```

Comment:
>public static AzureSupportCenterInsight CreateInsight<TResource>(Insight insight, OperationContext<TResource> context, Definition detector) [](start = 8, length = 139)

So technically this is exposed to detector writers. From what I understand, the platform is only interested in using this right?

Reply:
Yes, but since it is used from RuntimeHost.dll it has to be public right?

Reply:
Since this is technically a helper for the RuntimeHost I would move it there but because we don't have good discoverability in the detector editing experience you may not need to worry about it.

---
In reply to: [193864564](https://github.com/Azure/Azure-AppServices-Diagnostics/pull/39#discussion_r193864564) [](ancestors = 193864564)

Reply:
I am fine with that, though I am not that worried about someone accidentally using it. @ShekharGupta1988 thoughts? Only since you specifically created a project called models and utils haha

Reply:
A side affect of putting the utils there is that they all have to be public

Reply:
I like Hawk's Idea about moving this method to runtime host project. My motive of creating that Models/Utils dll was to expose functionality to detector authoring and sharing models between various projects. 
But specific helper methods which only one project is going to use can reside in that project itself.